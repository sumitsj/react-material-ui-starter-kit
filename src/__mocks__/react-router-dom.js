import React from 'react';
import * as PropTypes from 'prop-types';

export const Redirect = ({ to }) => <div title="redirect" to={to} />;

Redirect.propTypes = {
  to: PropTypes.string.isRequired,
};

export { HashRouter, Route, Link } from 'react-router-dom';

export default {};
