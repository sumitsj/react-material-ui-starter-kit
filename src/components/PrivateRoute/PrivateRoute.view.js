import React from 'react';
import * as PropTypes from 'prop-types';
import { Redirect, Route } from 'react-router-dom';

const PrivateRoute = ({ component: Component, ...rest }) => {
  const isAuthenticated = !!localStorage.getItem('auth-token');

  const render = (props) => {
    if (!isAuthenticated) {
      return <Redirect to="/login" />;
    }

    return <Component {...props} />;
  };

  return <Route {...rest} render={render} />;
};

PrivateRoute.propTypes = {
  component: PropTypes.elementType.isRequired,
};

export default PrivateRoute;
