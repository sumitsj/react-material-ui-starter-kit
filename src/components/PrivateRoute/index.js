import PrivateRoute from './PrivateRoute.view';

export default PrivateRoute;
