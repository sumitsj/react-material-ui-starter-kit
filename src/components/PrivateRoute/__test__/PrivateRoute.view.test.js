import React from 'react';
import { render } from '@testing-library/react';
import { HashRouter as Router, Route } from 'react-router-dom';
import PrivateRoute from '../PrivateRoute.view';

describe('Private Route', () => {
  const Home = () => 'Home Page';
  const Login = () => 'Login Page';

  const MockRouter = (
    <Router>
      <PrivateRoute path="/" component={Home} />
      <Route path="/login" component={Login} />
    </Router>
  );

  afterEach(() => {
    localStorage.getItem.mockRestore();
  });

  describe('HTML Structure', () => {
    it('should render passed component if authenticated', () => {
      localStorage.getItem.mockImplementation(() => 'true');
      const { asFragment } = render(MockRouter);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render Redirect component if NOT authenticated', () => {
      localStorage.getItem.mockImplementation(() => undefined);
      const { asFragment } = render(MockRouter);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
