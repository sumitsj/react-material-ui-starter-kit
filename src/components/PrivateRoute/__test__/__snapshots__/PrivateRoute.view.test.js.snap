// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Private Route HTML Structure should render Redirect component if NOT authenticated 1`] = `
<DocumentFragment>
  <div
    title="redirect"
    to="/login"
  />
</DocumentFragment>
`;

exports[`Private Route HTML Structure should render passed component if authenticated 1`] = `
<DocumentFragment>
  Home Page
</DocumentFragment>
`;
